frontend/model/contracts/group.js

```diff
@@ -459,7 +460,7 @@ sbp('chelonia/defineContract', {
       validate: objectMaybeOf({
         settings: objectMaybeOf({
           // TODO: add 'groupPubkey'
-          groupName: string,
+          groupName: stringMax(GROUP_NAME_MAX_CHAR, 'groupName'),
```

Comment:
I'd like to improve the security of the rest of the `string` values.

Let's take this opportunity to replace all `string` types in this file with `stringMax` types.

We can define a `MAX_HASH_LEN = 300` and use that for `manifestCid` and `txid`.

For `memo`, we can use `MAX_MEMO_LEN = 4096`.

`sharedValues` please use `GROUP_DESCRIPTION_MAX_CHAR` to match the UI limit.

`mincomeCurrency` you can use `10`.

`chatRoomID`, `toMemberID` and `proposalHash`, use `MAX_HASH_LEN`, and same for `proposalIds` (an array of `MAX_HASH_LEN` strings).

For `reason` the same as `GROUP_DESCRIPTION_MAX_CHAR` seems fine.

For the chatroom `name`, `CHATROOM_NAME_LIMITS_IN_CHARS`, and for chatroom `description`, `CHATROOM_DESCRIPTION_LIMITS_IN_CHARS`.

I think that's all of them. Please use your best judgement for anything I missed, and feel free to do the same for all of the validations in `chatroom.js` and `identity.js`.

The old manual-style validations in custom validation functions can be removed, and if there's nothing left inside of the custom validation function, the validator should be restored to using `objectOf`/`objectMaybeOf` directly.